packages/rule-components/src/RuleTable/RuleTable.js

```diff
@@ -0,0 +1,249 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Table, TableBody, TableHeader, sortable } from '@patternfly/react-table';
+import { Skeleton } from '@redhat-cloud-services/frontend-components/components/Skeleton';
+import { TableToolbar } from '@redhat-cloud-services/frontend-components/components/TableToolbar';
+import { Pagination, PaginationVariant } from '@patternfly/react-core/dist/js/components/Pagination';
+import { PrimaryToolbar } from '@redhat-cloud-services/frontend-components/components/PrimaryToolbar';
+import flatten from 'lodash/flatten';
+import debounce from 'lodash/debounce';
+import { calculateMeta, calculateActiveFilters, createRows } from './helpers';
+
+class RuleTable extends Component {
+    state = {
+        filterValues: {},
+        expanded: []
+    }
+
+    debounceFetchData = debounce(options => {
+        const { fetchData } = this.props;
+        fetchData(options);
+    }, 500)
+
+    componentDidUpdate(prevProps) {
+        if (JSON.stringify(this.props.filterValues) !== JSON.stringify(prevProps.filterValues)) {
```

Comment:
I understand that this solution will work, but this is not right way to comparing two objects, because it very sensitive on attribute order, so, please, I strongly suggest to rewrite it properly to some deepEqual function. Or use simple comparing: `this.props.filterValues !== prevProps.filterValues` because if nothing changed object pointer should be the same.

Reply:
I'll probably use `isEqual` from lodash, good point.